recipes/mbedtls/all/conanfile.py

```diff
@@ -90,9 +97,25 @@ def package(self):
         cmake = self._configure_cmake()
         cmake.install()
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "cmake"))
 
     def package_info(self):
-        # https://gitlab.kitware.com/cmake/cmake/blob/de7c21d677db1ddaeece03c19e13e448f4031511/CMakeLists.txt#L380
         self.cpp_info.names["cmake_find_package"] = "MbedTLS"
         self.cpp_info.names["cmake_find_package_multi"] = "MbedTLS"
-        self.cpp_info.libs = ["mbedtls", "mbedx509", "mbedcrypto"]
+        self.cpp_info.components["mbedcrypto"].names["cmake_find_package"] = "mbedcrypto"
+        self.cpp_info.components["mbedcrypto"].names["cmake_find_package_multi"] = "mbedcrypto"
+        self.cpp_info.components["mbedcrypto"].libs = ["mbedcrypto"]
+
+        self.cpp_info.components["mbedx509"].names["cmake_find_package"] = "mbedx509"
+        self.cpp_info.components["mbedx509"].names["cmake_find_package_multi"] = "mbedx509"
+        self.cpp_info.components["mbedx509"].libs = ["mbedx509"]
+        self.cpp_info.components["mbedx509"].requires = ["mbedcrypto"]
+
+        self.cpp_info.components["libembedtls"].names["cmake_find_package"] = "mbedtls"
+        self.cpp_info.components["libembedtls"].names["cmake_find_package_multi"] = "mbedtls"
+        self.cpp_info.components["libembedtls"].libs = ["mbedtls"]
+        self.cpp_info.components["libembedtls"].requires = ["mbedx509"]
+
+        if tools.Version(self.version) < "3.0.0" and self.options.with_zlib:
```

Comment:
It is hard to tell which library actually wants `zlib`.
On the CMakeLists they add `zlib` to a `${libs}` variable and then add that to every library they build, so I'm replicating that.
`ldd` doesn't show `zlib` on any of the three libs :confused:

Reply:
Sometimes libraries only check for existence of a library, but don't actually link to the library because they dynamically load it.
Then, it's important that the dependency is built as a shared library/module.
I don't know what is the case here.